### PR TITLE
[@types/simpl-schema] Add some missing types & convert `SimpleSchema` to a class

### PR DESCRIPTION
**Pull request by yurtsiv**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

In this PR I propose to add:

* Missing type for `autoValue`'s `this`  (https://github.com/aldeed/simpl-schema#autovalue). Note, that I didn't add all the fields from the documentation. See the issue: https://github.com/aldeed/simpl-schema/issues/412
* Missing fields in `CustomValidationContext`
* Some minor changes here and there + converted `SimpleSchemaStatic` interface to a class

There's still work to do, but I really need at least current changes to be merged.

Thanks in advance for the review!

### CODE REVIEW

File: types/simpl-schema/simpl-schema-tests.ts [hunk 1]
```diff
@@ -81,10 +121,7 @@ new SimpleSchema({
     shortInteger: SimpleSchema.Integer,
     shortDate: Date,
     shortArray: Array,
-    subSchema: StringSchemaWithOptions
+    subSchema: StringSchemaWithOptions,
 });
 
-SimpleSchema.extend(schema);
-SimpleSchema.extend(StringSchema);
```
Comment (yurtsiv) [inline types/simpl-schema/simpl-schema-tests.ts:88]: There's no static `extend` method

https://github.com/aldeed/simpl-schema/blob/main/package/lib/SimpleSchema.js#L477

**Comment by yurtsiv** [comment]
@arichter83 @deskoh @nicu-chiciuc @rafahoro 

Can someone take a look at this? Thanks!

**Review by arichter83 — COMMENTED** [review]
I will not review all formatting changes. Either you split your changes in two MRs for functionality and formatting, or you find somebody else.

**Comment by yurtsiv** [comment]
@arichter83 Now it should be easier to review. Will make a separate PR for formatting.

**Review by amcasey — CHANGES_REQUESTED** [review]
Looks like @yurtsiv has outstanding requests.

Edit: I need to read more carefully.  Sorry, @yurtsiv! 😆

**Review by amcasey — APPROVED** [review]
Seems fine to me.

**Comment by yurtsiv** [comment]
Ready to merge

**Comment by yurtsiv** [comment]
Ready to merge

**Pull request merged**